Understanding Ownership/Move Semantics/Mutable/src/main.rs

```diff
@@ -1,7 +1,7 @@
 fn main() {
     let vec0 = Vec::new();
 
-    let mut vec1 = fill_vec(vec0);
+    let vec1 = fill_vec(vec0);
```

Comment:
I get the [596 error](https://doc.rust-lang.org/error-index.html#E0596), as vec1 is not mutable anymore. Are we sure we do want to remove this?